### ISSUE TITLE
RTL not working with BTCPay for new users

### ISSUE THREAD

**Issue by NicolasDorier** [Ride-The-Lightning/RTL#1225]
**Describe the bug**

The following `RTL-config.json` is generated when starting RTL:

```json
{
  "port": "3000",
  "defaultNodeIndex": 1,
  "SSO": {
    "rtlSSO": 0,
    "rtlCookiePath": "",
    "logoutRedirectLink": ""
  },
  "nodes": [
    {
      "index": 1,
      "lnNode": "Node 1",
      "lnImplementation": "LND",
      "Authentication": {
        "macaroonPath": "/root/.lnd/data/chain/bitcoin/mainnet",
        "configPath": "/root/.lnd/lnd.conf"
      },
      "Settings": {
        "userPersona": "MERCHANT",
        "themeMode": "DAY",
        "themeColor": "PURPLE",
        "channelBackupPath": "/root/backup/node-1",
        "logLevel": "ERROR",
        "lnServerUrl": "https://127.0.0.1:8080",
        "fiatConversion": false,
        "unannouncedChannels": false
      }
    }
  ],
  "multiPass": "password"
}
```

In our setup, `/root/.lnd/data/chain/bitcoin/mainnet`, doesn't seem correct, it should be `/etc/lnd`.
What was working is:

```json
{
  "port": "3000",
  "defaultNodeIndex": 1,
  "SSO": {
    "rtlSSO": "0",
    "rtlCookiePath": "",
    "logoutRedirectLink": "/login"
  },
  "nodes": [
    {
      "index": 1,
      "lnNode": "Node 1",
      "lnImplementation": "LND",
      "Authentication": {
        "macaroonPath": "",
        "configPath": ""
      },
      "Settings": {
        "userPersona": "MERCHANT",
        "fiatConversion": false,
        "themeMode": "NIGHT",
        "themeColor": "INDIGO",
        "lnServerUrl": "https://localhost:8080/v1",
        "logLevel": "ERROR"
      }
    }
  ],
  "multiPassHashed": ""
}
```

Notice it is not the only property that is different from before.
~Also `"multiPass": "password"` appears to me problematic. If people are running this, does it effectively mean that everybody can connect to their RTL with password `password`~?

The situation is quite messy now as some of our users have the former config, and other have the later config.
How can we fix the situation for all?

**Comment by ShahanaFarooqui** [comment]
@NicolasDorier @saubyk 

This config change can't impact BTCPayServer setup. It sets up variables through environment and they always takes up higher priority over RTL-config.json.

In fact, I and @petzsch tested it today with new deployment and it worked as expected with the new RTL-Config.json. 

To clearify multipass value confusion, RTL will not accept any password if SSO is set, even if the user tries to provide it manually.

Regarding "unable to read lnd config" issue, we just had to change the variable name from 'LND_CONFIG_PATH` to `CONFIG_PATH` to fix it.

Keeping this open for now in case anything else needs to be addressed.

**Comment by strnly** [comment]
Not sure if related but here's what happened to me, with my CLN BTCPayServer 

I had an issue with Ride the Lighting and lost 8M+ sats opening a channel with your another peer,  from my node ID.

Last week, I tried opening a channel with another peer and it remained stuck on CHANNELD_AWAITING_LOCKIN for 4 days with no on-chain opening tx in the mempool. Yesterday I decided to SSH into the node and ran command `./bitcoin-lightning-cli.sh close $PEERID 10 # Force close after 10 second` to force close the channel since that was the only way according to guidance I could find. I even tweeted about it. I see the channel has now been closed but immediately after the it switched to _unilateral closing_ status, the opening on-chain transaction went through and the whole 8m sats were sent to the other node i was trying to open the channel with. These were supposed to be outbound liquidity but somehow, everything is gone from my node now.

https://mempool.space/tx/40a32072d541d7cc571253887e0550e3850a0fba2bf49b2664b306cec0014282

So basically, I lost 8M sats, which are likely on-chain on the other node. 

What are my options? Pretty messed up UX tbh.

**Comment by NicolasDorier** [comment]
@ShahanaFarooqui my apologize, you are right.
I was trying to reproduce the issue of one of our users and I jumped the gun on the cause of the problem.
I will dig, letting you know tomorrow.

**Comment by NicolasDorier** [comment]
@strnly I pretty much doubt you lost the money, and if you did I don't think it can be caused by RTL. Let's take it to mattermost to find out what happened.

**Comment by NicolasDorier** [comment]
Closing this one. I believe we have a problem when we update to the latest version but this is probably different matter than this issue.
